db/multi_db.go

```diff
@@ -0,0 +1,99 @@
+package db
+
+import (
+	"github.com/rakutentech/go-echo-kit/config"
+	"github.com/rakutentech/go-echo-kit/logger"
+
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+	gormlogger "gorm.io/gorm/logger"
+	"gorm.io/plugin/dbresolver"
+
+	"log"
+	"os"
+	"sync"
+	"time"
+)
+
+var onceMultiDb sync.Once
+var SingletonDB *gorm.DB
+
+type multiDbConf struct {
+	Master string // master db dsn
+	Slave  []string // slave db dsn
+	Name   string // db name
+}
+
+// return singleton DB
+func ConnDB(conf[] multiDbConf) *gorm.DB {
+	if len(conf) == 0 {
+		logger.LogCritf("[Fatal Error]can not connect to DB: empty dsn given")
+	}
+
+	onceMultiDb.Do(func() {
+		cfg := config.New()
+		env := cfg.GetString("app.env")
+
+		gormConfig := &gorm.Config{}
+
+		// print Slow SQL and happening errors
+		if env != "prod" && env != "pre" {
```

Comment:
app env check information don't need to be inside this enabler.
It should be enabled using envs like QUERY_LOG_ENABLED or SQL_LOGGER_ENABLED